app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -145,6 +157,25 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  private def getLogInputStream(zis: ZipInputStream, attemptTarget: WebTarget): (Option[InputStream], String) = {
+    // The logs are stored in a ZIP archive with a single entry.
+    // It should be named as "$logPrefix.$archiveExtension", but
+    // we trust Spark to get it right.
+    val entry = zis.getNextEntry
+    if (entry == null) {
+      logger.warn(s"failed to resolve log for ${attemptTarget.getUri}")
+      (None, "")
+    } else {
+      val entryName = entry.getName
+      if (entryName.endsWith(IN_PROGRESS)) {
```

Comment:
Nice! I had a quick look and it seems FS-based fetcher does not have the same logic. Could you double-check this please?